python/ray/ml/train/integrations/sklearn/sklearn_trainer.py

```diff
@@ -95,43 +120,303 @@ class SklearnTrainer(Trainer):
+
+            for name in test_scores:
+                results[key][f"test_{name}"] = test_scores[name]
+        return results
+
+    def _score_cv(
+        self,
+        estimator: BaseEstimator,
+        X: pd.DataFrame,
+        y: pd.Series,
+        groups: pd.Series,
+        n_jobs: int = 1,
+    ) -> Dict[str, Any]:
+        if not self.cv:
+            return {}
+
+        cv_results = cross_validate(
+            estimator=estimator,
+            X=X,
+            y=y,
+            groups=groups,
+            scoring=self.scoring,
+            cv=self.cv,
+            n_jobs=n_jobs,
+            fit_params=self.fit_params,
+            return_train_score=self.return_train_score_cv,
+        )
+        cv_aggregates = {}
+        for k, v in cv_results.items():
+            if not isinstance(v, np.ndarray):
+                continue
+            try:
+                cv_aggregates[f"{k}_mean"] = np.nanmean(v)
+                cv_aggregates[f"{k}_std"] = np.nanstd(v)
+            except Exception as e:
+                logger.warning(
+                    f"Couldn't calculate aggregate metrics for CV folds! {e}"
+                )
+                cv_aggregates[f"{k}_mean"] = np.nan
+                cv_aggregates[f"{k}_std"] = np.nan
+                pass
+
+        return {"cv": {**cv_results, **cv_aggregates}}
+
+    def _get_cv_parallelism(self, has_gpus: bool) -> bool:
+        parallelize_cv = False
+
+        assert not (has_gpus and self.parallelize_cv)
+
+        estimator_has_parallelism_params = has_cpu_params(self.estimator)
+
+        if self.cv and self.parallelize_cv is True:
+            parallelize_cv = True
+
+        if (
+            not has_gpus
+            and self.cv
+            and self.parallelize_cv is None
+            and not estimator_has_parallelism_params
+        ):
+            logger.info(
+                "No parallelism-related params detected in estimator, "
+                "will parallelize cross-validation instead."
+            )
+            parallelize_cv = True
+        return parallelize_cv
 
     def training_loop(self) -> None:
-        return super().training_loop()
+        register_ray()
+
+        self.estimator.set_params(**self.params)
+
+        datasets = self._get_datasets()
+        X_train, y_train = datasets.pop(TRAIN_DATASET_KEY)
+        groups = None
+        if "cv_groups" in X_train.columns:
+            groups = X_train["cv_groups"]
+            X_train = X_train.drop("cv_groups", axis=1)
+
+        scaling_config_dataclass = self._validate_and_get_scaling_config_data_class(
+            self.scaling_config
+        )
+
+        num_workers = scaling_config_dataclass.num_workers or 0
+        assert num_workers == 0  # num_workers is not in scaling config allowed_keys
+        has_gpus = scaling_config_dataclass.use_gpu
+
+        trainer_resources = scaling_config_dataclass.trainer_resources or {"CPU": 1}
+        num_cpus = trainer_resources.get("CPU", 1.0)
```

Comment:
```suggestion
        num_cpus = int(trainer_resources.get("CPU", 1)) 
```
Has to be an int, otherwise sklearn will throw an exception in some estimators